source/code/plugin/filter_cadvisor2mdm.rb

```diff
@@ -194,20 +199,21 @@ def get_metric_records(record, metric_name, metric_value, percentage_metric_valu
             return records
         end
 
-        
+
         def filter_stream(tag, es)
             new_es = MultiEventStream.new
-            ensure_cpu_memory_capacity_set
-            es.each { |time, record|
-              begin
+            begin
+                ensure_cpu_memory_capacity_set
+                es.each { |time, record|
                 filtered_records = filter(tag, time, record)
-                filtered_records.each {|filtered_record| 
+                filtered_records.each {|filtered_record|
                     new_es.add(time, filtered_record) if filtered_record
-                } if filtered_records
-              rescue => e
-                router.emit_error_event(tag, time, record, e)
-              end
-            }
+                    } if filtered_records
+                }
+            rescue => e
+                ApplicationInsightsUtility.sendExceptionTelemetry(e.backtrace)
```

Comment:
Not sure this will be useful? How could we query for exceptions from this filter plugin with this approach ? We are entirely relying on the stack to even identify these exceptions (and they could be different/empty every time). What we would need is a custom dimension identifying which plugin/place is the exception about or else append the exception message with plug-in name that would be more useful..

Reply:
I will add a task for this, and address that.